pf/internal/plugin/provider_server.go

```diff
@@ -80,5 +81,41 @@ func (p providerThunk) Configure(
 	if ctx.Value(setupConfigureKey) != nil {
 		return plugin.ConfigureResponse{}, nil
 	}
+	req.Inputs = removeSecrets(req.Inputs)
+	contract.Assertf(!req.Inputs.ContainsSecrets(),
+		"Inputs to configure should not contain secrets")
 	return p.GrpcProvider.Configure(ctx, req)
 }
+
+func removeSecrets(v resource.PropertyMap) resource.PropertyMap {
+	var remove func(resource.PropertyValue) resource.PropertyValue
```

Comment:
We need a function that can be recursive over `resource.PropertyValue` instead of `resource.PropertyMap`. It's an inner function because I don't want to pollute the scope for `plugin`.